onadata/apps/logger/models/xform.py

```diff
@@ -953,8 +953,17 @@ def soft_delete(self, user=None):
             update_fields.append('deleted_by')
 
         self.save(update_fields=update_fields)
+        # Delete associated filtered datasets
         for dataview in self.dataview_set.all():
             dataview.soft_delete(user)
+        # Delete associated Merged-Datasets
+        for merged_dataset in self.mergedxform_ptr.all():
+            merged_dataset.soft_delete(user)
+        # Delete associated Form Media Files
+        for metadata in self.metadata_set.all():
+            # Only target media metadata
+            if metadata.data_type == 'media':
+                metadata.delete()
```

Comment:
We should look into soft-deleting metadata perhaps? This seems like it would cause issues when the soft-deleted XForm is restored...

Reply:
I considered this, and I saw it wouldnt really be necessary.
Metadata holds form extras link enketo links and submission review enabling/disabling. These are often never deleted.
Its only now we are introducing deleting of media files associated with a form, and I dont believe we would want to restore these?, maybe have the user re-create them? I could be wrong

Reply:
I think it'll lead to a bad user experience... Users should be aware that deleting their data does have some consequences but yea I'm not entirely sure about this

Thoughts @denniswambua ?

Reply:
@WinnyTroy Not all metadata can be regenerated. So we will need the soft delete on metadata.

Reply:
@denniswambua  made this change